pkg/kubelet/volume/volume_manager.go

```diff
@@ -209,16 +213,42 @@ func (vm *volumeManager) GetVolumesForPodAndAppendSupplementalGroups(
 }
 
 func (vm *volumeManager) GetVolumesInUse() []api.UniqueVolumeName {
-	attachedVolumes := vm.actualStateOfWorld.GetAttachedVolumes()
-	volumesInUse :=
-		make([]api.UniqueVolumeName, 0 /* len */, len(attachedVolumes) /* cap */)
-	for _, attachedVolume := range attachedVolumes {
-		if attachedVolume.PluginIsAttachable {
-			volumesInUse = append(volumesInUse, attachedVolume.VolumeName)
+	// Report volumes in desired state of world and actual state of world so
+	// that volumes are marked in use as soon as the decision is made that the
+	// volume *should* be attached to this node until it is safely unmounted.
+	desiredVolumes := vm.desiredStateOfWorld.GetVolumesToMount()
+	mountedVolumes := vm.actualStateOfWorld.GetGloballyMountedVolumes()
+	volumesToReportInUse :=
+		make(
+			[]api.UniqueVolumeName,
+			0, /* len */
+			len(desiredVolumes)+len(mountedVolumes) /* cap */)
+	desiredVolumesMap :=
+		make(
+			map[api.UniqueVolumeName]bool,
+			len(desiredVolumes)+len(mountedVolumes) /* cap */)
+
+	for _, volume := range desiredVolumes {
+		if volume.PluginIsAttachable {
```

Comment:
SG, so just to confirm we won't end up waiting this 10s for all volumes like secrets etc right? just when a pod has an attachable volume (so CP backed volume)